larky/src/main/resources/vgs/http/request.star

```diff
@@ -0,0 +1,61 @@
+load("@stdlib//larky", larky="larky")
+load("@stdlib//urllib/request", urllib_request="request")
+
+
+def VGSHttpRequest(
+    url,
+    relative_url = None,
+    data=None,
+    headers={},
+    method=None
+):
+    super = urllib_request.Request(url)
+
+    self = super
+    self.__name__="VGSHttpRequest"
+    self.__class__=VGSHttpRequest
+
+    # relative_url property
+    def _get_relative_url():
+        return self._relative_url
+    def _set_relative_url(url):
+        self._relative_url = url
+    self.relative_url = larky.property(_get_relative_url, _set_relative_url)
+
+    # body property
+    def _get_body():
+        return self._data
+
+    def _set_body(data):
+        self.data = data
+    self.body = larky.property(_get_body, _set_body)
+
+    # override super
+    def add_header(key, val):
+        # original implementation uses key.capitalize(), however, we will not modify the keys.
+        self.headers[key] = val
+    self.add_header = add_header
+
+    # override super
+    def add_unredirected_header(key, val):
+        # will not be added to a redirected request
+        # original implementation uses key.capitalize(), however, we will not modify the keys.
+        self.unredirected_hdrs[key] = val
+    self.add_unredirected_header = add_unredirected_header
+
+    def __init__(
+        url,
+        relative_url = None,
+        data=None,
+        headers={},
+        method=None
+    ):
+        # call super init, with overrides
+        self.__init__(url, data=data, headers=headers, method=method)
```

Comment:
This does a reinitialization. You are aware of this ?

Reply:
yes, it's to invoke the overridden methods

Reply:
We should figure out a way to do this without re-init the class. This will do for now :P